### PR TITLE
Freehand point dropping for Polygon Tool

### PR DESCRIPTION
**Pull request by PaulHax**
Hold left mouse button down to keep dropping points.  Optimizes point count after polygon is "placed."

### CODE REVIEW

File: src/vtk/PolygonWidget/state.ts [hunk 1]
```diff
@@ -110,16 +116,43 @@ function vtkPolygonWidgetState(publicAPI: any, model: any) {
...
+    const imageMeta = getImageMetadata(tool.imageID);
+    const pixelSizeSquared = getPixelSizeSquared(
+      tool.frameOfReference,
+      imageMeta
+    );
+    if (pixelSizeSquared) {
+      const optimizedLine = decimate(
+        tool.points,
+        pixelSizeSquared * DECIMATE_PIXEL_SIZE_FACTOR
+      );
```
[the remark this answers is not in the document]
Comment (floryst) [reply src/vtk/PolygonWidget/state.ts:143]: What about using the smallest pixel dimension?

**Comment by floryst** [comment]
Freehand results in a lot of closely-spaced point handle circles, even with polyline decimation. One possible solution to this issue is to instead only show the closest handle circle to the mouse.

File: src/vtk/PolygonWidget/state.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (PaulHax) [reply src/vtk/PolygonWidget/state.ts]: Switched to smallest pixel dimension.

@manthey We are using Ramer–Douglas–Peucker algorithm to simplify the polyline.  Sort of interesting how to set the distance units epsilon param.

**Comment by PaulHax** [comment]
> Freehand results in a lot of closely-spaced point handle circles, even with polyline decimation. One possible solution to this issue is to instead only show the closest handle circle to the mouse.

Agreed.  Changed to showing handles on hover.

**Comment by floryst** [comment]
LGTM

**Pull request merged**